docs/extensibility/theme-support.md

```diff
@@ -8,7 +8,7 @@ To opt-in for one of these features, call `add_theme_support` in the `functions.
 
 ```php
 function mytheme_setup_theme_supported_features() {
-	add_theme_support( 'editor-color-palette',
+	add_theme_support( 'editor-color-palette', array(
```

Comment:
I've just added the array wrapper there because of [@jorgefilipecosta's comment](https://github.com/WordPress/gutenberg/pull/5273#discussion_r183658734)

Reply:
Can we map it to an ordered array before it gets exposed to JavaScript?

Reply:
@gziolo 
You mean like declaring the theme support with this:
`array(
	'color_name_1' => '#color1',
	'color_name_2' => '#color2',
)`
and then transform it in PHP to a JS required form of:
`array(
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
	),
)`
?

If so, this is certainly doable. But I'd like some more input on this, more opinions (maybe from @jorgefilipecosta)? You see, current associated array implementation is more self-explanatory in my opinion and could actually serve well in case of future extending of the palette color features.

Reply:
Hi,
Although I agree that it would be easier for themes to declare support using this syntax:
```
array(
	'color_name_1' => '#color1',
	'color_name_2' => '#color2',
)
````

I think it could limit future advances as the color value itself is a string and not an object we can expand.
For example, one possible improvement I was thinking about was to let themes define the contexts in which colors appear (e.g: allow some colors only for background).
In the current syntax, it is a matter of allowing an optimal property:

```
array(
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
		'context': => array( 'background' ),
	),
)
````

Reply:
Just like expected ;)